ui/components/src/Tooltip/utils.ts

```diff
@@ -58,3 +59,17 @@ export function assembleTransform(
     ? `translate3d(${x - cursorPaddingX}px, ${y}px, 0) translateX(-100%)`
     : `translate3d(${x + cursorPaddingX}px, ${y}px, 0)`;
 }
+
+/**
+ * Default to abbreviated value in y_axis but full value in tooltip
+ */
+export function getTooltipUnit(unit?: UnitOptions) {
+  if (!unit) {
+    return undefined;
+  }
+  const tooltipUnit = { ...unit };
+  if (tooltipUnit.kind === 'Decimal' && tooltipUnit.abbreviate) {
+    tooltipUnit.abbreviate = false;
```

Comment:
I thought about using immer, but decided not to since it isn't a required dependency in @perses-dev/components (at least not yet)